atom/feature_engineering.py

```diff
@@ -926,6 +951,80 @@ def check_y():
+                           "geneo": GeneticOptimization,
+                           "hho": HarrisHawkOptimization}
+
+            # initialization_params catches all params requied for zoofs algos
+            initialization_params = {"pso": ['n_iteration', 'timeout', 'population_size',
+                                             'minimize', 'c1', 'c2', 'w'],
+                                     "gwo": ['n_iteration', 'timeout', 'population_size',
+                                             'minimize', 'method'],
+                                     "dfo": ['n_iteration', 'timeout', 'population_size',
+                                             'minimize', 'method'],
+                                     "geneo": ['n_iteration', 'timeout',
+                                               'population_size', 'minimize',
+                                               'selective_pressure',
+                                               'elitism', 'mutation_rate'],
+                                     "hho": ['n_iteration', 'timeout', 'population_size',
+                                             'minimize', 'beta']}[self.strategy.lower()]
+
+            initialization_params_from_kwargs = {
+                k: v for k, v in self.kwargs.items() if k in initialization_params}
+            params_for_objective_function = {k: v for k, v in self.kwargs.items(
+            ) if k not in initialization_params + ['objective_function']}
+
+            if ("X_valid" in self.kwargs.keys()):
+                if ("y_valid" in self.kwargs.keys()):
+                    X_valid, y_valid = self._prepare_input(
+                        self.kwargs["X_valid"], self.kwargs["y_valid"])
+                    objective_function = custom_function_for_scorer
+                else:
+                    raise ValueError("Invalid value for the y_valid parameter. Value "
+                                     "cannot be absent in the presence of  X_valid.")
+
+            else:
+                X_valid, y_valid = X, y
+                objective_function = custom_cross_valid_function_for_scorer
+
+            # zoofs-native objective_function choice
+            if self.kwargs.get("objective_function"):
+                objective_function = self.kwargs['objective_function']
+
+            # atom-native scoring method can be chosen
+            elif self.kwargs.get("scoring"):
+                params_for_objective_function = {
+                    "scorer": get_custom_scorer(self.kwargs["scoring"])}
+
+            else:  # default scoring method is chosen if nothing is provided
+                params_for_objective_function = {"scorer":
+                                                 (get_custom_scorer("f1")
+                                                  if y.nunique() <= 2 else
+                                                     get_custom_scorer("f1_weighted"))
+                                                 if hasattr(self._solver,
```

Comment:
Not all classifiers have a `predict_proba` method. Instead use if `self.goal == "class" else`

Reply:
self.goal =='class' would work if solver is provided in atom native form like 'LGB_class' , is there any variable that tracks the task if only a model instance if provided by the user

Reply:
self.goal in FeatureSelection seems to be populated only when solver is an atom-native string format

Reply:
You are right. Use `if hasattr predict_proba or decision_function` then

Reply:
made the change in the latest commit